packages/client/src/sync/fetcher/storagefetcher.ts

```diff
@@ -119,20 +121,9 @@ export class StorageFetcher extends Fetcher<JobTask, StorageData[][], StorageDat
           slots[slots.length - 1].hash
         )}`
       )
-      for (let i = 0; i < slots.length - 1; i++) {
-        // ensure the range is monotonically increasing
-        if (bytesToBigInt(slots[i].hash) > bytesToBigInt(slots[i + 1].hash)) {
-          throw Error(
-            `Storage hashes not monotonically increasing: ${i} ${slots[i].hash} vs ${i + 1} ${
-              slots[i + 1].hash
-            }`
-          )
-        }
-      }
-      const trie = new Trie()
```

Comment:
i think just keep using the `_proofTrie` will cause memory blat as its db will keep getting build as more and more elemts get put to trie during verification

Reply:
Looking at the code for the `verifyProof` and `verifyRangeProof` functions, it seems like they create a new, disposable trie when they need to for checking proofs, so there shouldn't be bloat. Also see #2932 where the dedicated `_proofTrie` pattern was introduced in statemanager.